### ISSUE TITLE
Implement design elements for base email template

### ISSUE THREAD

**Issue by QaysarA** [department-of-veterans-affairs/notification-api#95]
### 1. Bold text
As a VANotify dev, I want to configure the base email template, so that specific text displays in bold. 

("Your claim details are below:", etc). Currently, Notify markup language doesn't support bolding (refer to CA Notify link in notes).

**AC:**
GIVEN I am configuring design in the base email template  
WHEN I add an element to display text as bold
THEN the text displays in bold.

### 2. Gray box/background
As a VANotify dev, I want to configure the base email template, so that a gray box/background displays behind specific text. 

This is replacing the current functionality for a header, in which a gray bar is shown.

**AC:**
GIVEN I am configuring design in the base email template  
WHEN I add an element to display a gray box/background behind the text
THEN a gray box/background displays behind the text.

Checklist:
- [x] Gray box
- [x] Bolding (`<strong>`)
- [x] Italicizing (`<em>`)

**Notes:** 
- Reference for template formatting and what Notify currently supports: https://notification.alpha.canada.ca/features/templates
- Refer to mockup for specific text that needs to be bolded and inside gray background: https://www.sketch.com/s/56d8d633-9bb2-4b88-a28a-3457250320d7#Inspector

**Comment by ffafara-tw** [comment]
**Dev analysis:**
- For the gray box, it should be enough to change styling in:
notifications_utils.formatters.NotifyEmailMarkdownRenderer.block_quote
Possible change border-left to background.

- For bolding:
Notification utils uses Mistune library for parsing markdown.
NotifyEmailMarkdownRenderer implements mistune.Renderer and overrides how certain elements are rendered.
In NotifyEmailMarkdownRenderer there are currently functions for emphasis and double_empahsis:
notifications_utils.formatters.NotifyEmailMarkdownRenderer.emphasis - it returns text wrapped in '*'
notifications_utils.formatters.NotifyEmailMarkdownRenderer.double_emphasis - it returns text wrapped in '**'
These might need to be modified to output html that will do the empahis / bolding. 
Please look at default implementation in mistune.py:
- mistune.Renderer.double_emphasis using <bold>
- mistune.Renderer.emphasis - using <em>

Another aspect to match a special character in template.
By default mistune uses (regex in mistune.py):
-  # __word__ or # **word** for double emphasis
- # _word_ or # *word* for emphasis 
Some of the regex patterns are overridden in formatters.py

Using ** or * does not currently work as it is matched against list item - probably due to the fact that list block regex is overwritten.
However using __ and _ works and is recognized as emphasis. Can be verified using template: 895713f9-61e2-41af-9707-b956e9029ac5 in dev environment.
So it should be enough to update:
- notifications_utils.formatters.NotifyEmailMarkdownRenderer.double_emphasis
- notifications_utils.formatters.NotifyEmailMarkdownRenderer.emphasis

**Comment by ffafara-tw** [comment]
Also very interesting perspective on using correct semantic html elements:
https://www.silkstream.net/blog/2016/02/b-vs-strong-i-vs-em-whats-the-difference.html

@toddstanich do you have suggestions wether we should be using <b> or <strong>
Should we ask Trevor Pierce about it as they might be handled differently by screen readers?

**Comment by toddstanich** [comment]
@ffafara-tw interesting find. I don't have a suggestion. Looking at VA.gov, `<strong>` appears to be used. I agree checking with Trevor would be best.

**Comment by marisahoenig** [comment]
Confirmed with Trevor to use `strong` since it is more semantic, as he saw no difference between `b` and `strong` when testing with a screen reader.

**Comment by marisahoenig** [comment]
Ready for desk check!

**Comment by marisahoenig** [comment]
Following desk check discussion, I tested a template with text in the middle of the block quote. I originally tested with only one line above and one line below, which just included the text in the block quote. When I added two lines before and two after, it separated it out.

Excerpt of template with 2 new lines:
```
123456\n\nFiller text here with two new lines.\n\n^ __Date submitted__
```
Behavior:
<img width="639" alt="Text with 2 lines" src="https://user-images.githubusercontent.com/13212345/87709612-86ccf680-c772-11ea-8647-c27632e44500.png">

Excerpt of template with 1 new line:
```
123456\nFiller text here with two new lines.\n^ __Date submitted__ 
```
Behavior:
<img width="634" alt="Text with 1 line" src="https://user-images.githubusercontent.com/13212345/87709913-fe028a80-c772-11ea-8945-df048f1b625f.png">
